pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -16,27 +16,27 @@
 from typing import Any, Dict, List, Union
 
 import torch.distributed as torch_distrib
+from torch.nn.parallel.distributed import DistributedDataParallel
 from torch.optim import Optimizer
 
 from pytorch_lightning import _logger as log
 from pytorch_lightning.core.lightning import LightningModule
-from pytorch_lightning.overrides.data_parallel import LightningDistributedDataParallel
+from pytorch_lightning.overrides.data_parallel import LightningDistributedModule, prepare_for_backward
 from pytorch_lightning.plugins.plugin import LightningPlugin
 
 
 class DDPPlugin(LightningPlugin):
     """
     Plugin to link a custom ddp implementation to any arbitrary accelerator.
 
-    This plugin forwards all constructor arguments to `LightningDistributedDataParallel`,
-    which in turn forwards all args to `DistributedDataParallel`.
+    This plugin forwards all constructor arguments to :class:`~torch.nn.parallel.DistributedDataParallel`.
 
     Example::
 
         class MyDDP(DDPPlugin):
 
             def configure_ddp(self, model, device_ids):
-                model = MyDDPWrapper(model, device_ids)
+                model = MyDDPWrapper(LightningDistributedModule(model), device_ids)
```

Comment:
question: what is MyDDPWrapper supposed to be here? wouldn't the user do now:

```python
class MyDDPModule(LightningDistributedModule):
    ...
```

and then use it in configure_ddp?

```python
            def configure_ddp(self, model, device_ids):
                model = MyDDPModule(model, device_ids)
```

Reply:
MyDDPWrapper can be a user's subclass of torch.nn.parallel.DistributedDataParallel. 

```python
class MyDDPWrapper(torch.nn.parallel.DistributedDataParallel):
    ...
```
    
and then the usage in configure_ddp is: 

```python
def configure_ddp(self, model, device_ids):
                model = MyDDPWrapper(LightningDistributedModule(model), device_ids)
```

The only requirement Lightning has is that the user wraps their model first with our LightningDistributedModule. That's all. I show the concrete example below in configure_ddp.

Reply:
By default MyDDPWrapper == torch.nn.parallel.DistributedDataParallel